apkid/rules/elf/obfuscators.yara

```diff
@@ -422,3 +422,20 @@ rule hikari : obfuscator
   condition:
     is_elf and all of them
 }
+
+rule dexprotector : obfuscator
+{
+  meta:
+    description = "DexProtector"
+    url         = "https://dexprotector.com/"
+    sample      = "d506e22003798f8b3a3d3c4a1b08af1cbd64667da6f9ed8cf73bc99ded73da44"
+    author      = "Eduardo Novella"
+
+  strings:
+    // - offset -   0 1  2 3  4 5  6 7  8 9  A B  C D  E F  0123456789ABCDEF
+    // 0x00000000  7f45 4c46 0201 0100 4450 4c46 00e0 0100  .ELF....DPLF....
+    $dp_elf_header = { 7f45 4c46 0201 0100 4450 4c46 }
```

Comment:
Good comments :D

Any idea with this DPLF thing? Since it's so early in the file it makes me think it could be fairly standard.

Reply:
**DexProtector Linkable Format** (DPLF) aka the proprietary format of ELF for DexProtector. This is my assumption only.

Reply:
the variable name specified the comment ;)  `dp_elf_header`

Reply:
```sh
[*] com.starlingbank.android.apkb!split_config.arm64_v8a.apk!lib/arm64-v8a/libdexprotector.so
 |-> obfuscator : DexProtector
[*] com.starlingbank.android.apkb!split_config.arm64_v8a.apk!lib/arm64-v8a/libnative-lib.so
 |-> obfuscator : DexProtector
```

Reply:
"DexProtector Linkable Format (DPLF)" is the key piece of information. Seems unique now. Could you add that as a comment? It's not obvious unless you're digging around DexProtector stuff all the time.

Reply:
Done